src/Storage/Notmuch.hs

```diff
@@ -218,6 +218,23 @@ threadToThread m = do
       <*> Notmuch.threadTotalMessages m
       <*> Notmuch.threadId m
 
+reloadTags
+    :: (MonadError Error m, MonadIO m, HasId a, ManageTags a)
+    => FilePath -> a -> m a
+reloadTags fp item = withDatabaseReadOnly fp go
+  where
+    go db = getThread db (getId item) >>= Notmuch.tags >>= pure . flip setTags item
```

Comment:
[nit] the `pure` is redundant.  you could do:
```haskell
go db = getThread db (getId item) >>= fmap (`setTags` item) . Notmuch.tags
```
or, for slightly better "ordering" i.e. operations all reading right-to-left instead of a mix:
```haskell
go db = fmap (`setTags` item) . Notmuch.tags =<< getThread db (getId item)
```

Reply:
Cheers. Done.